dataland-frontend/src/components/general/UserProfileDropDown.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <div @click="toggle" class="max-w-full max-h-full flex justify-content-center d-drop-down-toggle" name="profile-picture-dropdown-toggle">
+    <img ref="profile-picture" class="d-profile-picture" src="@/assets/images/logos/favicon-32x32.png" alt="User profile picture"/>
+    <img src="@/assets/images/elements/triangle_down.svg" class="d-triangle-down" alt="Open drop down menu icon"/>
+  </div>
+  <Menu ref="menu" :model="items" :popup="true" style="transform:translate(0px,1rem);" class="text-primary">
+    <template #item="{item}">
+      <a class="p-menuitem-link" role="menuitem" tabindex="0" @click="item.command()" :id="item.id"><span class="p-menuitem-icon text-primary" :class="item.icon"></span><span class="p-menuitem-text text-primary">{{item.label}}</span></a>
+    </template>
+  </Menu>
+</template>
+
+<script>
+import Menu from 'primevue/menu';
+export default {
+  name: "UserProfileDropDown",
+  inject: ['authenticated', 'getKeycloakInitPromise'],
+  components: {Menu},
+  methods: {
+    toggle(event) {
+      this.$refs.menu.toggle(event);
+    }
+  },
+  created() {
+    this.getKeycloakInitPromise().then((keycloak) => {
+      if (keycloak.authenticated && keycloak.idTokenParsed.picture) {
+        this.$refs["profile-picture"].src = keycloak.idTokenParsed.picture;
+      }
+    }).
+    catch((error) => console.log("error: " + error));
+  },
+  data() {
+    return {
+      items: [
+        {
+          label: 'Logout',
+          icon: 'pi pi-sign-out',
+          id: 'profile-picture-dropdown-toggle',
+          command: () => {
+            this.getKeycloakInitPromise().then((keycloak) => {
+              if (keycloak.authenticated) {
+                keycloak.logout()
+              }
+            }).catch((error) => console.log("error: " + error))
+          }
+        }
+      ]
+    }
+  }
+}
+</script>
+
+<style scoped>
+  .d-drop-down-toggle {
+    cursor: pointer;
+  }
+
+  .d-profile-picture {
+    border-radius: 50%;
+    height: 2rem;
+    /* TODO: Make the profile picture adjust size to fit parent automatically */
```

Comment:
Wir committen grundsätzlich keine TODOs und Kommentare nach main.
D.h. wir müssten das entweder direkt machen, oder in den Backlog schieben.